[gn] Add libcxx_enable_explicit_modules arg

atetubou opened the pull request:
This is to enable Clang's header modules build for libc++ instead of C++20's modules.

Currently, there are many errors due to missing includes when chromium tries to use explicit Clang modules build. e.g.
https://ci.chromium.org/ui/p/chromium/builders/ci/linux-modules-compile-fyi-rel/5368/overview

To find and fix them easily, I'd like to have GN args enabling explicit Clang modules build for libc++ in this repository too.

Some more context is in https://crbug.com/40263312.

atetubou commented:
@nico could you review this?

atetubou commented:
Sorry, this is still draft to use explicit clang modules.

nico commented:
lgtm

Please tweak the commit message to explicitly say that this is for explicit clang-style header modules, not C++20 style modules (…right)?

atetubou committed:
git squash commit for add_explicit_module_build_in_libxx.

8292a3c050309cff2e3a40db84bb231d06dfbe7d
git squash commit for add_explicit_module_build_in_libxx.

3dbdc43841dd89009fae759309008aea3cd35575
git squash commit for add_explicit_module_build_in_libxx.

e7b4cbb56fb0759508750eeca2cf1ecc6242e3cf
git squash commit for add_explicit_module_build_in_libxx.

b504ac5645c4fd0949dd39cbe913cc4d43afd003
[gn] Add libcxx_enable_explicit_modules arg

This is to enable Clang modules build for libc++.

Currently, there are many errors due to missing includes when
chromium tries to use explicit Clang modules build.
e.g.
https://ci.chromium.org/ui/p/chromium/builders/ci/linux-modules-compile-fyi-rel/5368/overview

To find and fix them easily, I'd like to have GN args enabling
explicit Clang modules build for libc++ in this repository too.

abef8470a575c8a6f00b9bd5d6eb7c6dbea7aa81
flat

1e9bbab918299369bda2de27f902fcb3fd776615
hoge

89445ac1c706ea9624b8cedb7704f18fb9f1fe5b
module

9b2b9ada760b7ada8bff644a687f6d5b79ae0c7c
config

32cb37f21edc149922d593b0f446e2cd63b63fa3
fmt

77ec3df9a68332af32e53ed920639790d0be7aed
clang

atetubou commented:
@nico I fixed configs to build libc++ with explicit modules enabled. Could you take a look at this PR again?

atetubou committed:
fmt

atetubou commented:
Per the discussion in https://crrev.com/c/6248376/6#message-0ddf8e6a0f3ce1eb1654f7025280d8ed75cf2e81, it is not necessary to use Clang modules for libc++ itself now.
